cpp/src/parquet/metadata.cc

```diff
@@ -851,23 +851,19 @@ class FileMetaDataBuilder::FileMetaDataBuilderImpl {
   }
 
   RowGroupMetaDataBuilder* AppendRowGroup() {
-    row_groups_.emplace_back(new format::RowGroup);
+    row_groups_.emplace_back();
     current_row_group_builder_ =
-        RowGroupMetaDataBuilder::Make(properties_, schema_, row_groups_.back().get());
+        RowGroupMetaDataBuilder::Make(properties_, schema_, &row_groups_.back());
```

Comment:
Taking the address of a vector element should be safe here since only the last element (current row group builder) is modified by the user at any time. The patch https://github.com/apache/arrow/pull/3261/files invalidated the previous row group builders.